core/types/transaction.go

```diff
@@ -367,6 +414,9 @@ func (tx *Transaction) Hash() common.Hash {
 	var h common.Hash
 	if tx.Type() == LegacyTxType {
 		h = rlpHash(tx.inner)
+	} else if tx.Type() == WithdrawalTxType {
+		// TODO: use SSZ root here instead?
+		h = prefixedRlpHash(tx.Type(), tx.inner)
```

Comment:
This appears to be a duplicate of the `else` block.  Why not just leave this `else if` condition out entirely?

Regarding the comment, you **SHOULD** include the transaction type in any signing (to avoid transaction malleability, see rationale for EIP-2718), and for future consistency I recommend applying that same rule for hashing in general (even when signatures aren't used) just so you can have consistency across the codebase.

Partially out of scope for this PR, and I suspect we have discussed this before, but why are typed transactions "RlpHashed"?  Why not just hash the bytes directly (no need for RLP, you already have a byte array so it seems you should just hash that byte array).

Reply:
i pulled it out to call out where we would use the SSZ hash tree root if we went down that route but thinking now to just use the typical `prefixedRlpHash` so i'll go remove now

yes agree, and that is what `prefixedRlpHash` does here to mix in the type to any hash (you could then turn around and sign)

Reply:
In the blob txs EIP SSZ type we define `prefixedSSZHash` that does `keccak256(type ++ hash_tree_root(tx.message))`.
This outer `keccak256` leaves compatibility with the proof structure of other transaction type pre-images, is collision resistant with other compatible SSZ types due to the type prefix, and exposes the `hash_tree_root` to do those fancy merkle proofs over. In this case I would do the same, but `tx.message` just becomes `tx`, since the transaction is not signed.